### ISSUE TITLE
"Install PyLint" Fails on first attempt

### ISSUE THREAD

**Issue by alexwhittemore** [DonJayamanne/pythonVSCode#933]
## Environment data
VS Code version:  1.11.2
Python Extension version:  0.6.3
Python Version:   2.7.13
OS and version:    Windows 10, 14393.1066

## Actual behavior
When, in a fresh environment, the user gets
`Error: PyLint is not installed` and clicks "Install PyLint", VSCode opens a terminal (PowerShell), but doesn't actually install PyLint. It requires invoking the linter a second time (i.e. by saving the file), incurring that error a second time, then clicking "InstallPyLint" a second time, before PyLint actually gets installed.
## Expected  behavior
Upon clicking "Install PyLint," the terminal should open and PyLint should install all in one go, without requiring a second pass.
## Steps to reproduce:  
-  Install a completely fresh python virtualenv, make a fresh project dir, and launch Code into this.
```
~ $ pip list
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a
 format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
dirsync (2.2.2)
pip (9.0.1)
setuptools (28.8.0)
six (1.10.0)
virtualenv (15.1.0)
virtualenvwrapper-powershell (12.7.8)
~ $ mkvirtualenv vscode-test
New python executable in C:\Users\alexw\.virtualenvs\vscode-test\Scripts\python.exe
Installing setuptools, pip, wheel...done.
Added activation script por Powershell to C:\Users\alexw/.virtualenvs/vscode-test\Scripts.
(vscode-test)~ $ mkdir C:\Users\alexw\Desktop\vscode-test


    Directory: C:\Users\alexw\Desktop


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----         5/4/2017  10:13 AM                vscode-test


(vscode-test)~ $ cd C:\Users\alexw\Desktop\vscode-test\
(vscode-test)vscode-test $ code .
```
- Create a new "MyStuff.py" file, ideally with linter errors for checking (missing docstrings, poorly named vars, whatever)
- Save this file, invoking the linter.
- Observe the error flow outlined above when the linter is found not to be installed.

## Settings  
Your launch.json (if dealing with debugger issues):  
```json
Not debugger related
```   
Your settings.json:   
```json
{
    "workbench.welcome.enabled": false,
    "workbench.colorTheme": "Tinacious Design - alexw",
    "workbench.iconTheme": "easy-icons"
}
```   
## Logs
Output from ```Python``` output panel  
```
```   
Output from ```Console window``` (Help->Developer Tools menu)
```
```

**Comment by alexwhittemore** [comment]
Additional info: I originally noticed and characterized in pythonVSCode 0.6.3, but have reproduced in the latest 0.6.4, same steps.

**Comment by DonJayamanne** [comment]
Please ensure the path to python interpreter is set in settings.json, then retry.

**Comment by alexwhittemore** [comment]
I'm using a virtualenv, and setting the environment variables for that by launching code from a powershell:
```
$ workon vscode-test
$ cd test-proj
$ code .
```

In any event, I did try modifying my settings.json as requested:
```
{
    "workbench.welcome.enabled": false,
    "workbench.colorTheme": "Tinacious Design - alexw",
    "workbench.iconTheme": "easy-icons",
    "python.venvPath": "~/.virtualenvs",
    "python.pythonPath": ".virtualenvs/vscode-test/Scripts/python.exe"
}
```

The same behavior persists - the user must "Install PyLint" a second time before it'll actually run the command in the built-in powershell.

A separate odd issue came up when I made those settings changes: suddenly I get the following error on launch
![capture](https://cloud.githubusercontent.com/assets/332827/25725609/a20fc1fc-30d5-11e7-8d89-f36c0beec1a5.PNG)

and test discovery fails.

**Comment by DonJayamanne** [comment]
If your ```.virtualenvs``` folder in the root directory, if not don't you need to prefix it with ```~/.virtualenvs```. Please try that.
I.e. please try the following path:
```json  
"python.pythonPath": "~/.virtualenvs/vscode-test/Scripts/python.exe"
```

**Comment by alexwhittemore** [comment]
Ahh, yes, excellent point. That does fix the "separate odd issue" with the error and test discovery failing.

It has no impact on the original issue. My environment gets set correctly by launching code from the powershell-with-venv-activated, setting it manually doesn't change any behavior (now that I'm doing so correctly).

**Comment by alexwhittemore** [comment]
Is the call to create a terminal on installer.ts line 136 blocking? It looks almost like it's not, so if a terminal doesn't exist yet, there's a race condition where the call farther down the file to send it the install command might silently fail because it hasn't come up yet. Powershell doesn't queue input while it's doing stuff like bash does, and it happens to be the case that mine takes a few ms to load.
https://github.com/DonJayamanne/pythonVSCode/blob/2a1f7f8f26ed6eaf2df7a4d2ef64186c0b37efab/src/client/common/installer.ts#L135-L137

**Comment by DonJayamanne** [comment]
@alexwhittemore Apologies for not getting back to you on this.
Please could you let me know if this issue has been resolved in a subsequent build.
If not, do let me know and I'll look into this. Once again, I do apologize for the delayed response.

**Comment by DonJayamanne** [comment]
Closing due to inactivity

**Comment by alexwhittemore** [comment]
Just verified, this issue still exists in the latest plugin version. Of course, it's low priority since any given environment will only ever encounter it once, and the workaround is "keep clicking when it asks."

tl;dr version: a powershell terminal must already be open in vscode when the "install pylint" button is clicked, or else the flow only gets as far as opening the requisite terminal (and then doesn't install pylint until you click again later).

When I start with a fresh virtualenv (no pylint installed yet) and manually open a terminal in vscode before clicking the "install pylint" prompt, it still fails. It has to be the "Python Installer" powershell.
